lms/djangoapps/lti_provider/views.py

```diff
@@ -84,7 +89,17 @@ def lti_launch(request, course_id, usage_id):
 
     # Create an edX account if the user identifed by the LTI launch doesn't have
     # one already, and log the edX account into the platform.
-    authenticate_lti_user(request, params['user_id'], lti_consumer)
+    lti_params = {}
+    for key in [LTI_PARAM_EMAIL, LTI_PARAM_FIRST_NAME, LTI_PARAM_LAST_NAME]:
+        if key in params:
+            lti_params[key] = params[key]
+        else:
+            lti_params[key] = None
+
+    authenticate_lti_user(request, params['user_id'], lti_consumer, 
+        lti_params[LTI_PARAM_EMAIL], lti_params[LTI_PARAM_FIRST_NAME], lti_params[LTI_PARAM_LAST_NAME])
+    if request.user.is_authenticated():
```

Comment:
I agree but in some sense it could be considered as a united task: we are adding lti-users with meaningful data (a email instead of a random username) in the edx gradebook.